src/main/java/org/isf/disctype/rest/DischargeTypeController.java

```diff
@@ -0,0 +1,107 @@
+import org.isf.disctype.model.DischargeType;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/dischargetype", produces = MediaType.APPLICATION_JSON_VALUE)
+public class DischargeTypeController {
+
+	@Autowired
+	protected DischargeTypeBrowserManager discTypeManager;
+
+	private final Logger logger = LoggerFactory.getLogger(DischargeTypeController.class);
+
+	public DischargeTypeController(DischargeTypeBrowserManager discTypeManager) {
+		this.discTypeManager = discTypeManager;
+	}
+
+	@PostMapping(value = "/dischargetypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> newDischargeType(@RequestBody DischargeTypeDTO admissionTypeDTO) throws OHServiceException {
+		String code = admissionTypeDTO.getCode();
+		logger.info("Create discharge type " + code);
+		boolean isCreated = discTypeManager.newDischargeType(getObjectMapper().map(admissionTypeDTO, DischargeType.class));
+		DischargeType admtCreated = null;
+		List<DischargeType> admtFounds = discTypeManager.getDischargeType().stream().filter(ad -> ad.getCode().equals(code))
+				.collect(Collectors.toList());
+		if (admtFounds.size() > 0)
+			admtCreated = admtFounds.get(0);
+		if (!isCreated || admtCreated == null) {
+			throw new OHAPIException(
+					new OHExceptionMessage(null, "discharge type is not created!", OHSeverityLevel.ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		return ResponseEntity.status(HttpStatus.CREATED).body(admtCreated.getCode());
+	}
+
+	@PutMapping(value = "/dischargetypes/{code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> updateDischargeTypet(@PathVariable String code,
+			@RequestBody DischargeTypeDTO admissionTypeDTO) throws OHServiceException {
+		logger.info("Update dischargetypes code:" + code);
+		DischargeType admt = getObjectMapper().map(admissionTypeDTO, DischargeType.class);
+		admt.setCode(code);
+		if(!discTypeManager.codeControl(code)) 
+			throw new OHAPIException(
+					new OHExceptionMessage(null, "discharge type not found!", OHSeverityLevel.ERROR));
```

Comment:
I think that the codeControl verify the presence of another entity with the same code. 
So it is not useful in an update context.